### PR TITLE
[SPARK-21891] [SQL] Add TBLPROPERTIES to DDL statement: CREATE TABLE USING

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
Add `TBLPROPERTIES` to the DDL statement `CREATE TABLE USING`. 

After this change, the DDL becomes
```
CREATE [TEMPORARY] TABLE [IF NOT EXISTS] [db_name.]table_name
USING table_provider
[OPTIONS table_property_list]
[PARTITIONED BY (col_name, col_name, ...)]
[CLUSTERED BY (col_name, col_name, ...)
 [SORTED BY (col_name [ASC|DESC], ...)]
 INTO num_buckets BUCKETS
]
[LOCATION path]
[COMMENT table_comment]
[TBLPROPERTIES (property_name=property_value, ...)]
[[AS] select_statement];
```

## How was this patch tested?
Add a few tests

### CODE REVIEW

**Comment by gatorsmile** [comment]
@cloud-fan

File: sql/core/src/test/resources/sql-tests/results/cross-join.sql.out [hunk 1]
```diff
@@ -128,6 +128,7 @@ two	2	two	2	one	1	two	2
 two	2	two	2	three	3	two	2
 two	2	two	2	two	2	two	2
 
+
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/results/cross-join.sql.out]: This was automatically added by`SQLQueryTestSuite`

File: sql/core/src/test/scala/org/apache/spark/sql/execution/OptimizeMetadataOnlyQuerySuite.scala [hunk 2]
```diff
@@ -117,4 +117,12 @@ class OptimizeMetadataOnlyQuerySuite extends QueryTest with SharedSQLContext {
     "select partcol1, max(partcol2) from srcpart where partcol1 = 0 group by rollup (partcol1)",
     "select partcol2 from (select partcol2 from srcpart where partcol1 = 0 union all " +
       "select partcol2 from srcpart where partcol1 = 1) t group by partcol2")
+
+  test("SPARK-21884 Fix StackOverflowError on MetadataOnlyQuery") {
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/OptimizeMetadataOnlyQuerySuite.scala]: Here, just add a test case from https://github.com/apache/spark/pull/19094 for verifying the fix in https://github.com/apache/spark/pull/18686

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/OptimizeMetadataOnlyQuerySuite.scala]: +1 for preventing futhre regressino.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by gatorsmile** [comment]
retest this please

**Review by jiangxb1987 — APPROVED** [review]
LGTM

**Pull request closed without merging**